flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -249,6 +266,66 @@ public static StateBackend fromApplicationOrConfigOrDefault(
         return backend;
     }
 
+    /**
+     * This is the state backend loader that loads a {@link DelegateStateBackend} wrapping the state
+     * backend loaded from {@link StateBackendLoader#fromApplicationOrConfigOrDefault} when
+     * delegation is enabled. If delegation is not enabled, the underlying wrapped state backend is
+     * returned instead.
+     *
+     * <p>{@link DelegateStateBackend} can only be enabled through configuration.
+     *
+     * @param fromApplication StateBackend defined from application
+     * @param config The configuration to load the state backend from
+     * @param classLoader The class loader that should be used to load the state backend
+     * @param logger Optionally, a logger to log actions to (may be null)
+     * @return The instantiated state backend.
+     * @throws DynamicCodeLoadingException Thrown if a state backend (factory) is configured and the
+     *     (factory) class was not found or could not be instantiated
+     * @throws IllegalConfigurationException May be thrown by the StateBackendFactory when creating
+     *     / configuring the state backend in the factory
+     * @throws IOException May be thrown by the StateBackendFactory when instantiating the state
+     *     backend
+     */
+    public static StateBackend loadStateBackend(
+            @Nullable StateBackend fromApplication,
+            Configuration config,
+            ClassLoader classLoader,
+            @Nullable Logger logger)
+            throws IllegalConfigurationException, DynamicCodeLoadingException, IOException {
+
+        final StateBackend backend =
+                fromApplicationOrConfigOrDefault(fromApplication, config, classLoader, logger);
+
+        if (config.get(CheckpointingOptions.ENABLE_STATE_CHANGE_LOG)) {
```

Comment:
Is a similar branch missing in `loadStateBackendFromConfig`?
It is called for example from `StreamExecutionEnvironment.loadStateBackend` which I think should get a wrapped version.